wsmaster/codenvy-hosted-api-audit/src/main/java/com/codenvy/api/audit/server/AuditReportPrinter.java

```diff
@@ -0,0 +1,175 @@
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.api.audit.server;
+
+import com.codenvy.api.license.CodenvyLicense;
+import com.codenvy.api.permission.server.model.impl.AbstractPermissions;
+
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.user.server.model.impl.UserImpl;
+import org.eclipse.che.api.workspace.server.model.impl.WorkspaceImpl;
+import org.eclipse.che.commons.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+import java.text.SimpleDateFormat;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.io.Files.append;
+
+/**
+ * Audit report printer.
+ *
+ * @author Igor Vinokur
+ */
+@Singleton
+class AuditReportPrinter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AuditReportPrinter.class);
+
+    /**
+     * Prints header of audit report in format:
+     * <p>
+     * Number of all users: <value>
+     * Number of users licensed: <value>
+     * Date when license expires: <date>
+     * <p>
+     * If license is not defined it prints error message instead license rows.
+     *
+     * @param auditReport
+     *         file of audit report
+     * @param allUsersNumber
+     *         all users number
+     * @param license
+     *         {@link CodenvyLicense} object
+     * @throws ServerException
+     *         if an error occurs
+     */
+    void printHeader(Path auditReport, long allUsersNumber, @Nullable CodenvyLicense license) throws ServerException {
+        printRow("Number of all users: " + allUsersNumber + "\n", auditReport);
+        if (license != null) {
+            printRow("Number of users licensed: " + license.getNumberOfUsers() + "\n", auditReport);
+            printRow("Date when license expires: " + new SimpleDateFormat("dd MMMM yyyy").format(license.getExpirationDate()) + "\n",
+                     auditReport);
+        } else {
+            printError("Failed to retrieve license", auditReport);
+        }
+    }
+
+    /**
+     * Print users info and his related workspaces info in format:
+     * <p>
+     * <user e-mail> is owner of <own workspaces number> workspace(s) and has permissions in <workspaces number> workspace(s)
+     * └ <workspace name>, is owner: <true or false>, permissions: <list of permissions for current user>
+     * .
+     * .
+     * .
+     * └ <workspace name>, is owner: <true or false>, permissions: <list of permissions for current user>
+     * <p>
+     * If the list of user workspaces is not defined it prints error message.
+     *
+     * @param auditReport
+     *         file of audit report
+     * @param user
+     *         {@link UserImpl} object
```

Comment:
I don't like comments like this because it says nothing about how this value will be used, it just says about parameter type and it just duplicates information from method's signature